lib/Dialect/AIE/IR/AIEDialect.cpp

```diff
@@ -1545,60 +1547,93 @@ LogicalResult DMABDOp::verify() {
   if (!isa<BufferOp, ExternalBufferOp>(getBuffer().getDefiningOp()))
     return emitOpError(
         "BDs only support BufferOp or ExternalBufferOp operands.");
-  if (auto memOp = getOperation()->getParentOfType<MemOp>()) {
+
+  if (getLenInBytes() % 4)
+    return emitOpError("transfer length must be multiple of 4B (32b)");
```

Comment:
nit: the parameter is just a number so it's a multiple of 4.  Or, don't abbreviate Bytes to B.